src/app/models/AnalysisJobItemResult.ts

```diff
@@ -44,6 +46,8 @@ export class AnalysisJobItemResult
   public readonly hasZip?: boolean;
   public readonly type?: ResultsItemType = "directory";
   public readonly children?: AnalysisJobItemResult[];
+  public parentItem?: AnalysisJobItemResult;
+  public open?: boolean;
```

Comment:
These properties represent view model properties. They have nothing to do with the model but rather how the model is presented. They should be in a different class.